control-plane/pkg/reconciler/trigger/trigger.go

```diff
@@ -63,6 +67,7 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, trigger *eventing.Trigge
 }
 
 func (r *Reconciler) reconcileKind(ctx context.Context, trigger *eventing.Trigger) reconciler.Event {
+	ctx = r.WithCtx(ctx)
```

Comment:
I've checked knative/eventing code and I see the feature store is used that way. But I found it strange. Why don't the feature store is set into the ctx instead of creating a wrapper function? Is it for testing?

Reply:
I am not "asking" you to change it, but I don't get why it is done that way in knative/eventing repo as well.

Reply:
@aliok I updated the approach and now there's a callback that gets triggered by the ConfigWatcher which just updates the cached feature flags in the reconciler. I removed the WithCtx.